### DISCUSSION TITLE
Apply button disabling on non completed date range selection

### DISCUSSION THREAD

**Discussion by protricker** [easepick/easepick#37] [Q&A] [answered]
Hello. 
I use DateRange plugin with manual date applying (autoApply:false). If i select start date i can click "Apply" button without selecting end date. Yes, in this case selected date will be empty, but I would like it to work like "litepicker". Apply button will be active when start date and end date were clicked.

Examples from lite picker:

Clicked only start date.
![image](https://user-images.githubusercontent.com/62597533/163204304-fd25ae3c-74c3-4ec7-b39a-88a2368ba1ab.png)

Full date range was selected
![image](https://user-images.githubusercontent.com/62597533/163203710-6470e44e-d771-48ec-b783-c24e443ab4a0.png)

I've tried to use ```picker.on('select', () => {})```  listener in setup to disable apply button when full range isn't selected. But can't get access to button element.

Could you suggest me how to do this?

**Comment by wakirin** [comment, marked as answer]
Good point.
Currently you can do something like this: https://jsfiddle.net/waki/2m4abL1x/
There is also a problem with style. I forget to add `:not(disabled)` for hover style.

I will fix it in the next release.

**Reply by protricker** [reply]
Thank you for answer. But there is a little problem. View event works only for first click to select start date. After end date selection 'view' is not called and apply button remains disabled. I found strange solution how to make apply button active after range selected by using 'click' event. But it works: https://jsfiddle.net/tvmaLe69/1/

**Reply by wakirin** [reply]
You are right. I prepared the example in a hurry and missed this thing.
Thanks for sharing solution.

**Comment by giorgiofnool** [comment]
With version 1.1.6 'element' is now 'target'
![image](https://user-images.githubusercontent.com/5812801/163960642-96434d5e-51b6-4b29-b953-154f3bb62424.png)

**Reply by wakirin** [reply]
You don't need the fix from the answer with version 1.1.6

I changed `element` to `target` to keep the details of the event consistent. (All other have `target`).

**Reply by giorgiofnool** [reply]
Sorry!
I was using the fix to make the button always active because I found that with RangePlugin strict false is not possible to select a range date. (I guess it's a bug or I misunderstood the option)
https://jsfiddle.net/4fevk9b8/

**Reply by wakirin** [reply]
You are right. I see that with `strict: false` the problem is reproduced.
I will prepare a fix for it.
Thanks for inform!

**Reply by giorgiofnool** [reply]
Thanks to you!

**Reply by wakirin** [reply]
fixed in 1.1.7

https://jsfiddle.net/waki/cuxe2vrn/
